doc/userguide/lua/libs/base64.rst

```diff
@@ -0,0 +1,55 @@
+Base64
+------
+
+Base64 functions are expose to Lua scripts with the
+``suricata.base64``. For example::
+
+  local base64 = require("suricata.base64")
+
+Functions
+~~~~~~~~~
+
+``encode(string)``
+^^^^^^^^^^^^^^^^^^
+
+Encode a buffer with standard base64 encoding. This standard encoding
+includes padding.
+
+``decode(string)``
+^^^^^^^^^^^^^^^^^^
+
+Decode a base64 string that contains padding.
```

Comment:
are the terms `string` and `buffer` considered interchangeable in this Base64 section?

Reply:
Yes, and I somewhat did that purposefully. In Lua the datatype is `string`, whether or not that data is a string of characters like a `char *`, or a binary blob of data `uint8_t *`.

Reply:
I guess one difference is that `encode` usually takes what you might call a buffer.  And `decode` always takes a string.